src/components/SideBar.tsx

```diff
@@ -70,20 +74,31 @@ function SideBar({
 	onChange,
 	...props
 }: SideBarInterface) {
-	const [modalOpen, setModalOpen] = useState(false);
+	const [courseModalOpen, setCourseModalOpen] = useState(false);
+	const [semesterModalOpen, setSemesterModalOpen] = useState(false);
+
 	const handleModalClose = () => {
-		setModalOpen(false);
+		courseModalOpen
+			? setCourseModalOpen(false)
+			: setSemesterModalOpen(false);
```

Comment:
Do you think there is ever a case where both could somehow be opened at the same time?